source/fundamentals/builders/aggregates.txt

```diff
@@ -11,3 +11,501 @@ Aggregates Builders
    :class: singlecol
 
 .. _aggregates-builders:
+
+This guide provides an overview of the :java-core-api:`Aggregates </com/mongodb/client/model/Aggregates>` class
+which provides static factory methods that build :manual:`aggregation pipeline
+operators </reference/operator/aggregation/>`.
+
+.. tip::
+
+   For brevity, you may choose to import the methods of the ``Aggregates`` class
+   statically:
+
+   .. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+      :start-after: begin static import
+      :end-before: end static import
+      :language: java
+      :dedent:
+
+   The examples on this page assume this static import.
+
+Match
+-----
+
+Use the ``match()`` method to create a :manual:`$match </reference/operator/aggregation/match/>`
+pipeline stage that matches incoming documents against the specified
+query filter, filtering out documents that do not match.
+
+.. tip::
+
+   The filter can be an instance of any class that implements ``Bson``, but it's
+   convenient to combine with use of the :ref:`Filters <filters-builders>` class.
+
+This example creates a pipeline stage that matches all documents where the
+``title`` field is equal to "The Shawshank Redemption":
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: begin match
+   :end-before: end match
+   :language: java
+   :dedent:
+
+
+
+Project
+-------
+
+Use the ``project()`` method to create a :manual:`$project </reference/operator/aggregation/project/>`
+pipeline stage that project specified document fields.
+
+.. tip::
+
+   Though the projection can be an instance of any class that implements ``Bson``,
+   it's convenient to combine with use of :ref:`Projections <projections-builders>`.
+
+This example creates a pipeline stage that excludes the ``_id`` field but
+includes the ``title`` and ``plot`` fields:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: begin project
+   :end-before: end project
+   :language: java
+   :dedent:
+
+Projecting Computed Fields
+++++++++++++++++++++++++++
+
+The ``$project`` stage can project computed fields as well.
+
+This example creates a pipeline stage that projects the ``rated`` field
+into a new field called ``rating``, effectively renaming the field.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/builders/AggBuilders.java
+   :start-after: begin computed
+   :end-before: end computed
+   :language: java
+   :dedent:
+
+
+Sample
+------
+
+Use the ``sample()`` method to create a :manual:`$sample </reference/operator/aggregation/sample/>`
+pipeline stage to randomly select documents from input.
```

Comment:
I opted for `input` because it's slightly more accurate in terms of aggregation (unless using match as the first stage, documents just flow in). Also, there's no restriction that says sample has to be the first stage, so you could use a sample stage after a match stage, for example.